R/lagwalk.R

```diff
@@ -391,3 +391,92 @@ model_sum.RW <- function(x) {
   }
   method
 }
+
+#' Refit a RW model
+#'
+#' Applies a fitted random walk model to a new dataset.
+#' 
+#' The models `NAIVE` and `SNAIVE` have no specific model parameters. Using `refit`
+#' for one of these models will provide the same estimation results as one would 
+#' use `fabletools::model(NAIVE(...))` (or `fabletools::model(SNAIVE(...))`. 
+#'
+#' @inheritParams refit.ARIMA
+#' @param reestimate If `TRUE`, the RW model will be re-estimated 
+#' to suit the new data. 
+#' 
+#' @examples
+#' lung_deaths_male <- as_tsibble(mdeaths)
+#' lung_deaths_female <- as_tsibble(fdeaths)
+#'
+#' fit <- lung_deaths_male %>%
+#'   model(RW(value ~ drift()))
+#'
+#' report(fit)
+#'
+#' fit %>%
+#'   refit(lung_deaths_female) %>%
+#'   report()
+#' @export
+refit.RW <- function(object, new_data, specials = NULL, reestimate = FALSE, ...) {
+  # Update data for re-evaluation
+  # update specials
+
+  specials$lag <- object$lag
+  if (!is_empty(object$b)) {
+    specials$drift <- object$b
+  }
+  
+  if (reestimate) {
+    return(train_lagwalk(new_data, specials, ...))
+  }
+  
+  # Case if reestimate = FALSE, only useful for RW with drift. 
+  if (!is_empty(object$b)) {
```

Comment:
Ideally `train_lagwalk()` would be updated to accept fixed parameters, from which `refit.RW()` can use to refit without re-estimation.